xds/internal/resolver/serviceconfig.go

```diff
@@ -142,6 +143,8 @@ type configSelector struct {
 }
 
 var errNoMatchedRouteFound = status.Errorf(codes.Unavailable, "no matched route was found")
+var errRouteActionUnsupported = status.Errorf(codes.Unavailable, "received route was of type RouteActionUnsupported, require type RouteActionRoute")
+var errRouteActionNonForwardingAction = status.Errorf(codes.Unavailable, "received route was of type RouteActionNonForwardingAction, require type RouteActionRoute")
```

Comment:
I would recommend getting rid of all these error sentinals and inlining them in the code.

Also, the error string for the errors added as part of this PR should be more like:
`matched route's action is type %s, only supported type is %s`.

You would also have to add a `String()` method to the `xdsresource.RouteActionType` enum, to satisfy the `Stringer` interface. See [this](https://go.dev/tour/methods/17) for details on the `Stringer` interface.